backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/CloneVmCommand.java

```diff
@@ -219,22 +232,24 @@ private void executeNextOperation() {
         }
     }
 
-    private Guid createVmSnapshot() {
+    private Guid getVmSnapshotToClone() {
         Snapshot activeSnapshot = snapshotDao.get(getSourceVmId(), Snapshot.SnapshotType.ACTIVE);
-        ActionReturnValue returnValue = runInternalAction(
-                ActionType.CreateSnapshotForVm,
-                buildCreateSnapshotParameters(),
-                ExecutionHandler.createDefaultContextForTasks(getContext()));
-
-        if (!returnValue.isValid()) {
-            getReturnValue().getValidationMessages().addAll(returnValue.getValidationMessages());
-            getReturnValue().setValid(false);
-            return null;
-        } else if (!returnValue.getSucceeded()) {
-            log.error("Failed to create VM snapshot");
-            throw new EngineException(returnValue.getFault().getError(), returnValue.getFault().getMessage());
+        // Skip VM snapshot creation for non-running VMs with MBS disks
+        if (isNoManagedBlockDiskAttachedToVm()) {
```

Comment:
Early return would be better here.

Reply:
Not sure about that...
On both cases (either there are no MBS disks or there are), I will use 'return activeSnapshot.getId();'
so there will be some code duplication.
@smelamud wdyt?

Reply:
It is not a significant duplication. But the code will be easier to read with an early return.

Reply:
Ack